### ISSUE TITLE
[CT-2555] Release and tag latest container images

### ISSUE THREAD

**Issue by dbeatty10** [dbt-labs/dbt-core#7571]
## Summary

Since 1.4.0, the `lastest` and `1.x.latest` tags for [ghcr Docker container image versions](https://github.com/dbt-labs/dbt-bigquery/pkgs/container/dbt-bigquery) have not been applied:
<img width="239" alt="image" src="https://github.com/dbt-labs/dbt-core/assets/44704949/30bb5ee0-66c6-4a3b-ad5c-9b54cb912069">

```[tasklist]
### Release the latest container images
- [ ] https://github.com/dbt-labs/dbt-core/issues/7149
- [ ] https://github.com/dbt-labs/dbt-core/issues/9597
- [ ] https://github.com/dbt-labs/dbt-bigquery/issues/1113
- [ ] https://github.com/dbt-labs/dbt-release/issues/109
```

```[tasklist]
### Tag latest container images
- [ ] https://github.com/dbt-labs/dbt-core/issues/7149
- [ ] https://github.com/dbt-labs/dbt-postgres/issues/9
- [ ] https://github.com/dbt-labs/dbt-snowflake/issues/597
- [ ] https://github.com/dbt-labs/dbt-bigquery/issues/941
- [ ] https://github.com/dbt-labs/dbt-redshift/issues/686
- [ ] https://github.com/dbt-labs/dbt-spark/issues/957
```

Here's a listing of the most recent tagged image of each:
- [dbt-postgres](https://github.com/dbt-labs/dbt-core/pkgs/container/dbt-postgres)
- [dbt-bigquery](https://github.com/dbt-labs/dbt-bigquery/pkgs/container/dbt-bigquery)
- [dbt-redshift](https://github.com/dbt-labs/dbt-redshift/pkgs/container/dbt-redshift)
- [dbt-snowflake](https://github.com/dbt-labs/dbt-snowflake/pkgs/container/dbt-snowflake)
- [dbt-spark](https://github.com/dbt-labs/dbt-spark/pkgs/container/dbt-spark)

**Comment by bdashrad** [comment]
It would also be nice to get semantic tagging, so you could pull a tag tag like `1.5` that would get the latest `1.5.x` tag.

**Comment by raginjason** [comment]
Seems like this is at least related to or would be fixed by https://github.com/dbt-labs/dbt-core/issues/7149

**Comment by dbeatty10** [comment]
Re-opening until we decided how to handle each of the remaining items in the tasklist.

Resolved #7149 instead in the meantime.
